exercises/practice/word-count/tests/word_count.cairo

```diff
@@ -0,0 +1,230 @@
+
+    let expected = array![
+        WordResult { word: "first", count: 1 },
+        WordResult { word: "laugh", count: 1 },
+        WordResult { word: "then", count: 1 },
+        WordResult { word: "don't", count: 2 },
+        WordResult { word: "cry", count: 1 },
+        WordResult { word: "you're", count: 1 },
+        WordResult { word: "getting", count: 1 },
+        WordResult { word: "it", count: 1 },
+    ]
+        .span();
+    assert_unordered(output, expected);
+}
+
+#[test]
+#[ignore]
+fn with_quotations() {
+    let input = "Joe can't tell between 'large' and large.";
+    let mut output = count_words(input);
+
+    let expected = array![
+        WordResult { word: "joe", count: 1 },
+        WordResult { word: "can't", count: 1 },
+        WordResult { word: "tell", count: 1 },
+        WordResult { word: "between", count: 1 },
+        WordResult { word: "and", count: 1 },
+        WordResult { word: "large", count: 2 }
+    ]
+        .span();
+    assert_unordered(output, expected);
+}
+
+#[test]
+#[ignore]
+fn substrings_from_the_beginning() {
+    let input = "Joe can't tell between app, apple and a.";
+    let mut output = count_words(input);
+
+    let expected = array![
+        WordResult { word: "joe", count: 1 },
+        WordResult { word: "can't", count: 1 },
+        WordResult { word: "tell", count: 1 },
+        WordResult { word: "between", count: 1 },
+        WordResult { word: "app", count: 1 },
+        WordResult { word: "apple", count: 1 },
+        WordResult { word: "and", count: 1 },
+        WordResult { word: "a", count: 1 }
+    ]
+        .span();
+    assert_unordered(output, expected);
+}
+
+#[test]
+#[ignore]
+fn multiple_spaces_not_detected_as_a_word() {
+    let input = " multiple   whitespaces";
+    let mut output = count_words(input);
+
+    let expected = array![
+        WordResult { word: "multiple", count: 1 }, WordResult { word: "whitespaces", count: 1 }
+    ]
+        .span();
+    assert_unordered(output, expected);
+}
+
+#[test]
+#[ignore]
+fn alternating_word_separators_not_detected_as_a_word() {
+    let input = ",\n,one,\n ,two \n 'three'";
+    let mut output = count_words(input);
+
+    let expected = array![
+        WordResult { word: "one", count: 1 },
+        WordResult { word: "two", count: 1 },
+        WordResult { word: "three", count: 1 }
+    ]
+        .span();
+    assert_unordered(output, expected);
+}
+
+#[test]
+#[ignore]
+fn quotation_for_word_with_apostrophe() {
+    let input = "can, can't, 'can't'";
+    let mut output = count_words(input);
+
+    let expected = array![
+        WordResult { word: "can", count: 1 }, WordResult { word: "can't", count: 2 }
+    ]
+        .span();
+    assert_unordered(output, expected);
+}
+
+
+// helper function.
+fn assert_unordered(span1: Span<WordResult>, span2: Span<WordResult>) {
+    for item in span1 {
+        for other_item in span2 {
+            if item.word == other_item.word {
```

Comment:
Think about what this function actually does:
1. Iterate `span1`
2. Iterate `span2`
3. If the `span1` item's word equals `span2` item's word, then:
    1. Assert that `span1` item's word equals `span2` item's word (**again**).
    2. Assert that `span1` item's count equals `span2` item's count.

This function ignores spans that have different word items. The only time it _does_ panic is if words match, but their counts don't.
You could say you got lucky that your implementation works despite the tests being unreliable. 

The below implementation is more useful:

```rust
fn assert_unordered(span1: Span<WordResult>, span2: Span<WordResult>) {
    for item in span1 {
        let mut found = false;
        for other_item in span2 {
            if item == other_item {
                found = true;
                break;
            }
        };
        assert!(
            found,
            "assertion failed: `(left == right)`\n  left: `{:?}`,\n  right: `{:?}`",
            span1,
            span2
        );
    }
}
```